Std/Data/List/Lemmas.lean

```diff
@@ -206,6 +229,66 @@ theorem zipWith_get? {f : α → β → γ} :
     | nil => simp
     | cons b bs => cases i <;> simp_all
 
+@[simp]
+theorem zipWith_eq_nil_iff {f : α → β → γ} {l l'} : zipWith f l l' = [] ↔ l = [] ∨ l' = [] := by
+  cases l <;> cases l' <;> simp
+
+theorem lt_length_of_lt_zipWith_left_length {f : α → β → γ} {i : Nat} {l : List α} {l' : List β}
+    (h : i < (zipWith f l l').length) : i < l.length := by
+  rw [length_zipWith, Nat.lt_min] at h
+  exact h.left
+
+theorem lt_length_of_lt_zipwith_right_length {f : α → β → γ} {i : Nat} {l : List α} {l' : List β}
+    (h : i < (zipWith f l l').length) : i < l'.length := by
+  rw [length_zipWith, Nat.lt_min] at h
+  exact h.right
```

Comment:
Sorry for delayed review.

It seems like it would be better to just make a lemma `(zipWith f l l').length = min l.length l'.length` and do the same for similar theorems such as `lt_length_left_of_zip`.

Reply:
I just deleted these theorems -- we indeed already have `length_zipWith` and `length_zip`.